libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1624,7 +1627,7 @@ void SMTEncoder::arrayPush(FunctionCall const& _funCall)
 
 void SMTEncoder::arrayPop(FunctionCall const& _funCall)
 {
-	auto memberAccess = dynamic_cast<MemberAccess const*>(&_funCall.expression());
+	auto memberAccess = dynamic_cast<MemberAccess const*>(cleanExpression(_funCall.expression()));
```

Comment:
If I understand your question correctly, then my explanation would be the following.
The early bailout in tuple expressions is just for SMTChecker to *not* try to create an SMT representation for expressions corresponding to function identifiers inside 1-tuple, such as `(data.pop)` from example.
The change you refer to is needed to get to the `data.pop` from the FunctionCall expression `(data.pop)()`, and then to the array expression `data` and its SMT representation.
This would be the place where the code previously crashed, because `_funCall.expression()` would return the `Tuple` expression `(data.pop)` not a `MemberAccess` expression. We now use the `cleanExpression` method to strip away some syntactic structures (mainly these nested 1-tuples) to get to Expression that we expect.